src/core/crypto/multiKey.ts

```diff
@@ -0,0 +1,122 @@
+import { Hex } from "../hex";
+import { HexInput } from "../../types";
+import { Deserializer } from "../../bcs/deserializer";
+import { Serializer } from "../../bcs/serializer";
+import { AnyPublicKey } from "./anyPublicKey";
+import { AnySignature } from "./anySignature";
+import { PublicKey } from "./asymmetricCrypto";
+
+export class MultiKey extends PublicKey {
+  /**
+   * List of any public keys
+   */
+  public readonly publicKeys: AnyPublicKey[];
+
+  /**
+   * The minimum number of valid signatures required, for the number of public keys specified
+   */
+  public readonly signaturesRequired: number;
+
+  constructor(args: { publicKeys: PublicKey[]; signaturesRequired: number }) {
+    super();
+    const { publicKeys, signaturesRequired } = args;
+
+    // Validate number of public keys is greater than signature required
+    if (signaturesRequired < 0) {
+      throw new Error("The number of required signatures needs to be greater then 0");
+    }
+
+    // Validate number of public keys is greater than signature required
+    if (publicKeys.length < signaturesRequired) {
+      throw new Error(
+        `Provided ${publicKeys.length} public keys is smaller than the ${signaturesRequired} required signatures`,
+      );
+    }
+
+    const keys: AnyPublicKey[] = [];
+    publicKeys.forEach((publicKey) => {
+      if (publicKey instanceof AnyPublicKey) {
+        keys.push(publicKey);
+      } else {
+        // if public key is instance of a legacy authentication key, i.e
+        // Legacy Ed25519, convert it into AnyPublicKey
+        keys.push(new AnyPublicKey(publicKey));
+      }
+    });
```

Comment:
Tbh, this is probably confusing for a user form the outside.  PublicKey vs AnyPublicKey.  We may want to work on naming.  Similarly I saw elsewhere with transaction types that gets somewhat confusing

Reply:
we use `AnyPublicKey` to reflect and match the Rust code. I explain it in different places and tbh, I think it is confusing no matter what so we need to make sure we document it well.

https://github.com/aptos-labs/aptos-ts-sdk/pull/141/files#diff-45edb2a64043b75aa2bf8796e392178277847b4e02a18ffae66a5adca7779a6fR9

https://github.com/aptos-labs/aptos-ts-sdk/pull/141/files#diff-819f43565d33e1eb1e2e53882e6bca77220613ee111ade4500b72862b102e2eeR14

https://github.com/aptos-labs/aptos-ts-sdk/pull/141/files#diff-98942ba717d1ba7eafd0b4cfa834f6307d611d3ba525aa2ae68a93938b97e0b5R14